gwt-localizable-processor/src/main/java/org/gwtproject/i18n/processor/LocalizableProcessingStep.java

```diff
@@ -47,105 +55,378 @@ public LocalizableProcessingStep(ProcessingEnvironment processingEnv) {
+    String enclosingElementName =
+        !(ElementKind.PACKAGE == element.getEnclosingElement().getKind())
+            ? element.getEnclosingElement().getSimpleName().toString() + "_"
+            : "";
+    String factoryClassName =
+        enclosingElementName + element.getSimpleName().toString() + "_Factory";
+
+    TypeSpec.Builder builder =
+        TypeSpec.classBuilder(factoryClassName).addModifiers(Modifier.PUBLIC);
+
+    MethodSpec.Builder createMethodBuilder =
+        MethodSpec.methodBuilder("create")
+            .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+            .returns(TypeName.get(element.asType()));
+
+    localeResourceMap
+        .keySet()
+        .stream()
+        .sorted(Comparator.reverseOrder())
+        .forEach(
+            locale -> {
+              createMethodBuilder
+                  .beginControlFlow("if($S.equals(System.getProperty($S)))", locale, "locale")
+                  .addStatement(
+                      "return new $T()",
+                      ClassName.bestGuess(
+                          context.elements.getPackageOf(element).getQualifiedName().toString()
+                              + "."
+                              + getConstantClassName(element, localePostfix(locale))))
+                  .endControlFlow();
+            });
+
+    createMethodBuilder.addStatement("return null");
+    builder.addMethod(createMethodBuilder.build());
+
+    writeSource(builder, context, element);
+  }
+
+  public Properties loadPropertiesByLocale(
+      Properties properties, AptContext context, Element element, String locale) {
+    TypeElement typeElement = (TypeElement) element;
+    List<? extends TypeMirror> interfaces = typeElement.getInterfaces();
 
-                  impl.addStatement("return $S", method.getSimpleName().toString());
+    if (!interfaces.isEmpty()) {
+      interfaces
+          .stream()
+          .map(context.types::asElement)
+          .forEach(
+              extendedInterface -> {
+                Properties propertiesResource = getPropertiesResource(extendedInterface, locale);
+                if (nonNull(propertiesResource)) {
+                  propertiesResource.forEach(properties::putIfAbsent);
                 }
 
-                defaultImplBuilder.addMethod(impl.build());
+                loadPropertiesByLocale(properties, context, extendedInterface, locale);
               });
+    }
 
-      try {
-        JavaFile.builder(packageName, defaultImplBuilder.build())
-            .build()
-            .writeTo(processingEnv.getFiler());
-      } catch (IOException e) {
-        e.printStackTrace();
+    return properties;
+  }
+
+  private void generateConstantsClasses(
+      Element element, AptContext context, Map<String, Properties> localeResourceMap) {
+
+    localeResourceMap.forEach(
+        (locale, resource) -> {
+          TypeSpec.Builder constantImplBuilder =
+              TypeSpec.classBuilder(getConstantClassName(element, locale))
```

Comment:
todo: add a Generated annotation, or at least a comment, since we expect libraries to ship with generated code...